segm8/segm8.py

```diff
@@ -0,0 +1,358 @@
+"""The module contains the SegM8 class that allows you to interact with
+a chain of a particular SegM8 7-segment indicator modules.
+This module requires that `spidev` be installed within the Python
+environment you are using it.
+"""
+
+import spidev
+from . import font
+from .output_format import Align, NumberSystem
+from .segm8_exceptions import DataOutOfWidth, OutOfDeviceRange
+
+
+class SegM8:
+
+    """Class for SegM8 7-segment module."""
+
+    def __init__(self, pin_CE, device_count=1):
+        """The constructor for SegM8 class.
+
+        Parameters:
+        -----------
+        pin_CE: int
+            The chip enable (also known as chip select or slave select)
+            pin used to control the shift-register latch. It can take
+            the values 0 or 1, which corresponds to the CE0 or CE1 pins
+            on the Raspberry Pi board.
+        device_count: int
+            The number of SegM8 modules connected in a daisy-chain. If
+            omitted, defines a single module.
+        """
+        self._pin_CE = pin_CE
+        self._device_count = device_count
+        # Set the internal buffer to the "dark" state.
+        self._data = [font.FONT[" "]] * self._device_count
+        # Create an SPI object on the "0" SPI bus and connected to
+        # "pin_CE".
+        self._spi = self._spi_init(0, pin_CE)
+
```

Comment:
Shouldn't you call _send_data at the end of the constructor to clear the indicators?

Reply:
Yes thanks, missed that. Fixed in the next commit.